bootstrap/eks/api/v1beta2/eksconfig_types.go

```diff
@@ -24,6 +24,9 @@ import (
 
 // EKSConfigSpec defines the desired state of Amazon EKS Bootstrap Configuration.
 type EKSConfigSpec struct {
+	// NodeType specifies the type of node (e.g., "al2023")
+	// +optional
+	NodeType string `json:"nodeType,omitempty"`
```

Comment:
This is only required when using AL2023 so perhaps let's add an enum for this and only accept `al2023`? Something like this:

```suggestion
// +kubebuilder:validation:Enum=al2023
type NodeType string

const (
	NodeTypeAL2023 = "al2023
)

// EKSConfigSpec defines the desired state of Amazon EKS Bootstrap Configuration.
type EKSConfigSpec struct {
	// NodeType specifies the type of node (e.g., "al2023")
	// +optional
	NodeType NodeType `json:"nodeType,omitempty"`
```